src/main/kotlin/com/atlassian/performance/tools/workspace/api/git/GitRepo.kt

```diff
@@ -29,14 +31,28 @@ interface GitRepo {
             if (descendant.isDirectory) {
                 descendant
                     .listFiles()
-                    .singleOrNull { it.name == ".git" }
-                    ?.let { return FileRepository(it) }
+                    ?.singleOrNull { it.name == ".git" }
+                    ?.let { return FileRepository(toGitFolder(it)) }
             }
-            val parent = descendant.parentFile
-            return when (parent) {
+            return when (val parent = descendant.parentFile) {
                 null -> null
                 else -> findInAncestors(parent)
             }
         }
+
+        private fun toGitFolder(file: File): File? {
+
+            if (file.isDirectory) {
+                return file
+            }
+
+            val line = file.bufferedReader().use { it.readLine() }
+            val pattern = "gitdir: (.*)".toRegex()
+            val match = pattern.find(line)
+            val path = match!!.groupValues[1]
+            val gitFolderPath = path.substring(0, path.indexOf(".git") + 4)
```

Comment:
Alternative without magic numbers:
```suggestion
            val gitFolderPath = path.replaceAfter(".git", "")
```
or
```suggestion
            val gitFolderPath = path.replaceAfter(".git", replacement = "")
```

Reply:
I could quickly experiment with micro refactoring thanks to the awesome unit tests! Much appreciated 🙏